tests/core/pyspec/eth2spec/test/helpers/fork_transition.py

```diff
@@ -0,0 +1,142 @@
+import random
+
+from eth2spec.test.helpers.state import (
+    next_slot,
+    state_transition_and_sign_block,
+    transition_to,
+)
+from eth2spec.test.helpers.block import (
+    build_empty_block_for_next_slot,
+    build_empty_block,
+    sign_block,
+)
+
+
+def _state_transition_and_sign_block_at_slot(spec,
+                                             state,
+                                             operation_dict=None):
```

Comment:
note: this function has been modified to accept an `operation_dict` with a batch of operations to ensure are included in the produced block